### PR TITLE
refactor(tools): optimize `utils.yield` for better performance

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

* Do not use division in `yield`.
* move `ngx.sleep`/ `ngx.get_phase`

### CODE REVIEW

File: kong/tools/utils.lua [hunk 1]
```diff
@@ -1434,21 +1432,28 @@ function _M.sort_by_handler_priority(a, b)
...
+
+    -- phase == "init" or phase == "init_worker"
+    if byte(phase, 1) == CHAR_I and byte(phase, 2) == CHAR_N then
       return
     end
     if in_loop then
-      counter = counter + 1
-      if counter % YIELD_ITERATIONS ~= 0 then
+      counter = counter - 1
+      if counter > 0 then
```
Comment (ADD-SP) [inline kong/tools/utils.lua:1449]: I think it is a nice change, I also did a test.

```lua
local random = math.random

local COUNTER = 1e5

local function test_sub()
    local n = COUNTER

    while n > 0 do
        n = n - 1
    end

    return random()
end

local function test_mod()
    local n = 0

    while n % COUNTER ~= 0 do
        n = n + 1
    end

    return random()
end

ngx.update_time()
local start = ngx.now()

test_sub()

ngx.update_time()
print(string.format("test_sub(): %f s", ngx.now() - start))

ngx.update_time()
start = ngx.now()

test_mod()

ngx.update_time()
print(string.format("test_mod(): %f s", ngx.now() - start))
```

```shell
$ resty test.lua
test_sub(): 12.196000 s
test_mod(): 34.310000 s
```

File: kong/tools/utils.lua [hunk 2]
```diff
@@ -1434,21 +1432,28 @@ function _M.sort_by_handler_priority(a, b)
...
+  local counter = YIELD_ITERATIONS
   function _M.yield(in_loop, phase)
     if ngx.IS_CLI then
       return
     end
     phase = phase or get_phase()
-    if phase == "init" or phase == "init_worker" then
+
+    -- phase == "init" or phase == "init_worker"
+    if byte(phase, 1) == CHAR_I and byte(phase, 2) == CHAR_N then
```
[the remark this answers is not in the document]
Comment (chronolaw) [reply kong/tools/utils.lua]: Thank you. The test for performance is very important.

File: kong/tools/utils.lua (hunk 1 above)
Comment (chronolaw) [reply kong/tools/utils.lua:1449]: Great job!

**Review by bungle — APPROVED** [review]

File: kong/tools/utils.lua [hunk 3]
```diff
@@ -1434,7 +1432,10 @@ function _M.sort_by_handler_priority(a, b)
 end
 
 do
-  local counter = 0
+  local get_phase      = ngx.get_phase
+  local ngx_sleep      = ngx.sleep
```
Comment (bungle) [inline kong/tools/utils.lua:1436]:
```suggestion
  local get_phase = ngx.get_phase
  local ngx_sleep = ngx.sleep
```

**Review by bungle — CHANGES_REQUESTED** [review]

**Review by bungle — APPROVED** [review]

**Pull request merged**